### ISSUE TITLE
Some things that do not work for me.

### ISSUE THREAD

**Issue by Itay1787** [thenickdude/KVM-Opencore#12]
So I was able to install everything that seems to work.
Just a few things I could not make work.

I am using Proxmox version 6.3-2
And I also passthrough my GT 710 for the MacOS VM and it works just fine.
But I also need to passthrough the ports in USB 3 and I can't make it work, I don't wont to transfer all the controller because some things in USB are shared with other VMs. So I want to passthrough the ports I need, If I transfer them in the form of USB 2 then the MacOS recognizes them and everything is fine.
But when I set to in the VM that the USB port are in USB 3 the MacOS no longer works. Seeing the Apple logo, it starts loading and then turns off the VM completely.
What to do? I need USB 3 and I can not buy a controller just for this VM.

Another thing I failed to do, I want to transfer my iGPU to MacOS
i5 7600k
hd 630
I need it for hardware decoding...

Hope I was understood. English is not my main language

Thanks
Itay

**Comment by thenickdude** [comment]
macOS doesn't seem to support Proxmox's emulated USB 3 controller.

For iGPU passthrough see this guide:

https://github.com/vivekmiyani/OSX_GVT-D

**Comment by Itay1787** [comment]
> macOS doesn't seem to support Proxmox's emulated USB 3 controller, or at least I've never had it working before. You'll need to untick the USB3 tickbox in the USB passthrough config.

Yeah I know,
If I do not click and try to passthrough something that is connected in USB 3 then it does not appear in MacOS... is there a way to get around this? Make USB 3 work. Change something in Proxmox or something ...?

> For iGPU passthrough see this guide

And I'll try it on my iGPU, thanks.

**Comment by thenickdude** [comment]
>Change something in Proxmox or something ...?

More likely it is possible to get the emulated XHCI controller working in macOS by adding kexts or OpenCore config.

**Comment by Itay1787** [comment]
> > Change something in Proxmox or something ...?
> 
> More likely it is possible to get the emulated XHCI controller working in macOS by adding kexts or OpenCore config.

Ok, what kexts should I add or change in openCore? I have already tried USBInjectAll and XHCI-unsupported both did not help.

**Comment by thenickdude** [comment]
No idea

**Comment by Itay1787** [comment]
> For iGPU passthrough see this guide:
> 
> https://github.com/vivekmiyani/OSX_GVT-D

Missing files… 
The file opregion.bin are missing. Did you try this before?

**Comment by thenickdude** [comment]
No, because I don't have an iGPU

**Comment by Itay1787** [comment]
Hello again, something very strange that I do not understand what is going on...
My sound card is supported on MacOS using AppleALC and in boot arguments adding alcid=11 I know this because I tried to install MacOS on the hardware itself and everything works. 
And when I try to pass the sound card to VM and put the same settings in the config, it doesn't work, MacOS does not recognize my sound card.

And in addition to that my motherboard has a USB controller that should be supported on MacOS without issue. But it doesn't work here either.
I see that MacOS has detected the USB 3 Hub but when I connect something it doesn't work. And of course I did both for the USB controller and for the sound card a PCi passhrough for the MacOS VM
Something very strange ... I do not know why it is like that.

**Comment by thenickdude** [comment]
It's likely that macOS and AppleALC are expecting the sound card to have specific names in the DSDT or whatever file it is, to match the names used by actual motherboards, you may have to edit that.

**Comment by Itay1787** [comment]
> It's likely that macOS and AppleALC are expecting the sound card to have specific names in the DSDT or whatever file it is, to match the names used by actual motherboards, you may have to edit that.

Ok, I didn't use a custom file , I used the files that OpenCore already provides in their guide.
And could it be the same reason why the USB controller is not working properly?

**Comment by thenickdude** [comment]
Doubtful, since my 3 USB controllers were plug and play. Are you getting any warnings in dmesg at VM start time about the USB controller failing to reset?

**Comment by Itay1787** [comment]
> Are you getting any warnings in dmesg at VM start time about the USB controller failing to reset?

No, VM and MacOS start fine. Chances are I did not notice, is there anything in particular to pay attention to?

**Comment by thenickdude** [comment]
Literally anything appearing in dmesg at VM start is interesting, just post all of those.

**Comment by Itay1787** [comment]
> Literally anything appearing in dmesg at VM start is interesting, just post all of those.

So ... the VM just decided to stop working. Tomorrow I will do a reinstall for everything (maybe it will fix things...) will keep updating here.
But as for the sound card what exactly do I need to do?

**Comment by thenickdude** [comment]
No idea

**Comment by Itay1787** [comment]
> Literally anything appearing in dmesg at VM start is interesting, just post all of those.

I did a reinstall and the audio still does not work

I have attached the Output of the boot from the terminal 
[https://mega.nz/file/dNRm0Tba#9AKS-RVt7g0bUH4ane_V5V5dKQDki4CdhKrQZc0fvbM](url)

**Comment by thenickdude** [comment]
I found out that the emulated USB 3 controller can be fixed by adding `-global nec-usb-xhci.msi=off` to the QEMU commandline
